azure-iot-device/tests/common/test_http_transport.py

```diff
@@ -0,0 +1,279 @@
+        assert mock_ssl_context_constructor.call_args == mocker.call(protocol=ssl.PROTOCOL_TLSv1_2)
+        assert mock_ssl_context.check_hostname is True
+        assert mock_ssl_context.verify_mode == ssl.CERT_REQUIRED
+
+    @pytest.mark.it(
+        "Configures TLS/SSL context using default certificates if protocol wrapper not instantiated with a CA certificate"
+    )
+    def test_configures_tls_context_with_default_certs(self, mocker):
+        mock_ssl_context = mocker.patch.object(ssl, "SSLContext").return_value
+
+        HTTPTransport(hostname=fake_hostname)
+
+        assert mock_ssl_context.load_default_certs.call_count == 1
+        assert mock_ssl_context.load_default_certs.call_args == mocker.call()
+
+    @pytest.mark.it(
+        "Configures TLS/SSL context with provided CA certificates if protocol wrapper instantiated with a CA certificate"
+    )
+    def test_configures_tls_context_with_ca_certs(self, mocker):
+        mock_ssl_context = mocker.patch.object(ssl, "SSLContext").return_value
+
+        HTTPTransport(hostname=fake_hostname, ca_cert=fake_ca_cert)
+
+        assert mock_ssl_context.load_verify_locations.call_count == 1
+        assert mock_ssl_context.load_verify_locations.call_args == mocker.call(cadata=fake_ca_cert)
+
+    @pytest.mark.it("Configures TLS/SSL context with client-provided-certificate-chain like x509")
+    def test_configures_tls_context_with_client_provided_certificate_chain(self, mocker):
+        fake_client_cert = X509("fantastic_beasts", "where_to_find_them", "alohomora")
+        mock_ssl_context_constructor = mocker.patch.object(ssl, "SSLContext")
+        mock_ssl_context = mock_ssl_context_constructor.return_value
+
+        HTTPTransport(hostname=fake_hostname, x509_cert=fake_client_cert)
+
+        assert mock_ssl_context.load_default_certs.call_count == 1
+        assert mock_ssl_context.load_cert_chain.call_count == 1
+        assert mock_ssl_context.load_cert_chain.call_args == mocker.call(
+            fake_client_cert.certificate_file,
+            fake_client_cert.key_file,
+            fake_client_cert.pass_phrase,
+        )
+
+
+class HTTPTransportTestConfig(object):
+    @pytest.fixture
+    def mock_http_client_constructor(self, mocker):
+        mocker.patch.object(ssl, "SSLContext").return_value
+        mocker.patch.object(HTTPTransport, "_create_ssl_context").return_value
+        mock_client_constructor = mocker.patch.object(http_client, "HTTPSConnection", autospec=True)
+        mock_client = mock_client_constructor.return_value
+        response_value = mock_client.getresponse.return_value
+        response_value.status = 1234
+        response_value.reason = "__fake_reason__"
+        response_value.read.return_value = "__fake_response_read_value__"
+        return mock_client_constructor
+
+    @pytest.fixture
+    def transport(self, mock_http_client_constructor):
+        return HTTPTransport(hostname=fake_hostname)
```

Comment:
This isn't strictly necessary here like it is on Stage `.run_op()` and `.handle_event()` tests. You could just define these fixtures on the test class below.